src/html5/parser/document.rs

```diff
@@ -683,11 +724,8 @@ impl TreeBuilder for DocumentHandle {
     fn insert_attribute(&mut self, key: &str, value: &str, element_id: NodeId) -> Result<()> {
```

Comment:
Ideally I'd like to fetch the node and do the existence/type checks here and pass the node to the other insert methods by reference but couldn't figure out how to properly do it without the borrow checker yelling at me so there's a little bit of code dupe. Open to suggestions